packages/components/src/public/QueryModel/CustomizeGridViewModal.tsx

```diff
@@ -113,6 +142,37 @@ export const CustomizeGridViewModal: FC<Props> = memo(props => {
         return columnsInView.findIndex(col => col === column) !== -1;
     }
 
+    const onDropField = useCallback((dropResult: DropResult): void => {
+        const { destination, draggableId, source } = dropResult;
+        if (destination === null || source.index === destination.index) {
+            return;
+        }
+        const { index } = destination;
+
+        const colInMotion = columnsInView[source.index];
+        let updatedColumns = columnsInView.filter(col => col.index != draggableId);
+        updatedColumns = [...updatedColumns.slice(0, index), colInMotion, ...updatedColumns.slice(index) ];
+        setColumnsInView(updatedColumns);
+        if (source.index === selectedIndex) {
+            setSelectedIndex(index);
+        } else if (selectedIndex !== undefined) {
+            if (source.index > selectedIndex && index <= selectedIndex) {
+                setSelectedIndex(selectedIndex + 1);
+            } else if (source.index < selectedIndex && index >= selectedIndex) {
+                setSelectedIndex(selectedIndex - 1);
+            }
+        }
+        setIsDirty(true);
+    }, [selectedIndex, columnsInView]);
+
+    const onSelectField = useCallback((index): void => {
+        if (index === selectedIndex) {
```

Comment:
This is a scenario where the state setters providing the value is very useful. This could be rewritten as:

```tsx
const onSelectField = useCallback((index: number): void => {
    setSelectedIndex(_selectedIndex => {
        if (_selectedIndex === index) return undefined;
        return index;
    });
}, []);
```

This will result in the `onSelectField` being effectively constant due to memoization. Same concept applies to the `addColumn` case above.

Reply:
Thanks for the suggestion. I'll make this change in the next branch since I'm on the verge of merging.

Reply:
FYI, I have applied this suggested change to fb_customizeViewLookups. Thanks for the suggestion, Nick.